sdk/core/azure-core/src/test/java/com/azure/android/core/common/TestUtils.java

```diff
@@ -1,29 +1,141 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.azure.android.core.common;
+package com.azure.android.storage.blob;
+
+import com.azure.android.core.http.ServiceClient;
+import com.azure.android.storage.blob.StorageBlobAsyncClient;
+import com.azure.android.storage.blob.StorageBlobClient;
+import com.azure.android.storage.blob.credential.SasTokenCredential;
+import com.azure.android.storage.blob.interceptor.SasTokenCredentialInterceptor;
+
+import org.threeten.bp.OffsetDateTime;
 
 import java.io.PrintWriter;
 import java.io.StringWriter;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
 import java.util.List;
 import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
 
+import okhttp3.Headers;
 import okhttp3.Interceptor;
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
 import okhttp3.mockwebserver.MockWebServer;
 
+import static org.junit.Assert.assertFalse;
+
 public final class TestUtils {
     private TestUtils() {
         // Empty constructor to prevent instantiation of this class.
     }
 
+    // ----------------------- CONSTANT VALUES ----------------------------
+    /*
+    The values below are used to create data-driven tests for access conditions.
+     */
+    static final OffsetDateTime oldDate = OffsetDateTime.now().minusDays(1);
+
+    static final OffsetDateTime newDate = OffsetDateTime.now().plusDays(1);
+
+
+    // -------------------- GENERATING CLIENTS --------------------------
+    public static StorageBlobClient.Builder initializeDefaultSyncBlobClientBuilder(Interceptor ... interceptors) {
+        return initializeDefaultSyncBlobClientBuilder(enableFiddler(), interceptors);
+    }
+
+    public static StorageBlobClient.Builder initializeDefaultSyncBlobClientBuilder(boolean enableFiddler,
+                                                                                   Interceptor ... interceptors) {
+        return new StorageBlobClient.Builder(getServiceBuilderWithOptionalProxy(enableFiddler, interceptors))
+            .setBlobServiceUrl(getDefaultEndpointString(useHttps()))
+            .setCredentialInterceptor(new SasTokenCredentialInterceptor(new SasTokenCredential(getDefaultSasToken())));
+    }
+
+    public static StorageBlobAsyncClient.Builder initializeDefaultAsyncBlobClientBuilder(Interceptor ... interceptors) {
+        return initializeDefaultAsyncBlobClientBuilder(enableFiddler(), interceptors);
+    }
+
+    public static StorageBlobAsyncClient.Builder initializeDefaultAsyncBlobClientBuilder(boolean enableFiddler,
+                                                                                         Interceptor ... interceptors) {
+        return new StorageBlobAsyncClient.Builder(UUID.randomUUID().toString(),
```

Comment:
can we replace all instances of UUID.randomUUID() with a method call? That way when we get to the point where we record, we can just change that method?

Reply:
Same idea with date time now

Reply:
Done